commands/deploy_test.go

```diff
@@ -42,6 +49,152 @@ func Test_deploy(t *testing.T) {
 	}
 }
 
+func Test_overrideFlag_deployToNamespace(t *testing.T) {
```

Comment:
There's so much going on in this test, I have a feeling that it will be hard for newcomers (or old-timers) to maintain. Have you got any suggestions for increasing the granularity?